### PR TITLE
Add SearchClient.batch_delete_by_subject

### PR DESCRIPTION
**Pull request by sirosen**
Exactly what the title says. :slightly_smiling_face: 

----
:books: Documentation preview :books:: https://globus-sdk-python--760.org.readthedocs.build/en/760/

### CODE REVIEW

File: src/globus_sdk/services/search/client.py [hunk 1]
```diff
@@ -397,6 +397,62 @@ def delete_by_query(
...
+                    sc = globus_sdk.SearchClient(...)
+                    query_data = {
+                        "q": "user query",
+                        "filters": [
+                            {
+                                "type": "range",
+                                "field_name": "path.to.date",
+                                "values": [{"from": "*", "to": "2014-11-07"}],
+                            }
+                        ],
```
Comment (derek-globus) [inline src/globus_sdk/services/search/client.py:429]: Is this relevant to the usage example?

File: src/globus_sdk/services/search/client.py [hunk 2]
```diff
@@ -397,6 +397,62 @@ def delete_by_query(
...
+        """
+        Delete data in a Search index as an asynchronous task, deleting multiple
+        documents based on their ``subject`` values.
+
+        A ``task_id`` value will be included in the response.
+
+        :param index_id: The index in which to delete data
+        :type index_id: str or UUID
+        :param data: a batch-delete-by-subject document listing documents to delete
+        :type data: dict
```
Comment (derek-globus) [inline src/globus_sdk/services/search/client.py:412]: I think we should accept a list of subjects (list of strings) explicitly here (with additional data params to keep in line with other operations).

No need to force people to look at additional docs to figure out what the post data object needs to look like for something so straightforward.

**Review by derek-globus — CHANGES_REQUESTED** [review]

File: src/globus_sdk/services/search/client.py (hunk 1 above)
Comment (sirosen) [reply src/globus_sdk/services/search/client.py]: I just was about to comment. I pushed an amend to remove this. It was a copy-pasta mistake.

Comment (derek-globus) [reply src/globus_sdk/services/search/client.py]: Any idea where I submit a request to have github include a pasta emoji in their list of reaction emojis?

File: src/globus_sdk/services/search/client.py (hunk 2 above)
Comment (sirosen) [reply src/globus_sdk/services/search/client.py:412]: Yeah, I'm okay with that as an interface change.

I'll do that and also expand the test to make an assertion about the payload sent.

Comment (sirosen) [reply src/globus_sdk/services/search/client.py]: I think you will like the new tests. It's nothing to `snork` home about, but it gets the job done.

I'm going to mark this as resolved, since the basic thrust of it (allowing `subjects=...` as usage) is covered.

**Review by derek-globus — APPROVED** [review]
